src/api/controller.js

```diff
@@ -306,7 +306,8 @@ export default class ApiController {
         qb
           .whereIn('id', likes)
           .orWhereIn('type', ['hashtag_like', 'school_like', 'geotag_like'])
-          .orderBy('posts.created_at', 'desc')
+          .andWhere('user_id', user_id[0])
```

Comment:
this doesn't look like a correct solution to me. you're limiting posts by author. which is not correct.

we need a union of 2 queries here:

1. old one which selects by id-list (likes)
2. new one which has filter by type and your condition

will you be able to add tests here?